featuretools/entityset/entityset.py

```diff
@@ -1034,14 +1044,20 @@ def add_last_time_indexes(self, updated_dataframes=None):
                             lti_df['last_time_old'] = lti_df['last_time_old'].astype('datetime64[ns]')
                             lti_df = lti_df.fillna(pd.to_datetime('1800-01-01 00:00')).max(axis=1)
                             lti_df = lti_df.replace(pd.to_datetime('1800-01-01 00:00'), pd.NaT)
-                    # lti_df = lti_df.apply(lambda x: x.dropna().max(), axis=1)
 
-                    dataframe.ww.metadata['last_time_index'] = lti_df
-                    dataframe.ww.metadata.get('last_time_index').name = 'last_time'
+                    es_lti_dict[dataframe.ww.name] = lti_df
+                    es_lti_dict[dataframe.ww.name].name = 'last_time'
 
             explored.add(dataframe.ww.name)
-        self.reset_data_description()
 
+        # Store the last time index on the DataFrames
+        for df in self.dataframes:
+            lti = es_lti_dict[df.ww.name]
+            if lti is not None:
+                df.ww['last_time'] = lti
```

Comment:
This introduces nans any time the underlying index of the dataframe doesn't match the underlying index of `lti`. For pandas, this shouldn't be an issue because it handles differently ordered indexes well, and the underlying index will always match the woodwork index.

But because dask and koalas never set the underlying index of their dataframes, this means that unless the indexes happen to match, there will be nans. (this is most easily seen with `es.set_last_time_index(['products'])` since products has a categorical index. 

I tried mapping `lti`'s underlying index to the woodwork index, but that proved to be messy, and I'm not entirely sure possible with dask. I also tried using just the underlying index throughout last time index calculation, but because of the `groupby` on the woodwork index values, we'd still need a way of mapping to the actual underlying index.

Reply:
Solved by using `merge` to combine the dataframe with the last time index. This creates a new DataFrame object for Dask and Koalas. Pandas still uses the setitem, so the object is the same.